nix/configuration.nix

```diff
@@ -0,0 +1,307 @@
+    users.users.jelle = {
+      isNormalUser = true;
+      description = "Jelle Besseling";
+      extraGroups = [ "wheel" ];
+      openssh.authorizedKeys.keys = [
+        "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICunYiTe1MOJsGC5OBn69bewMBS5bCCE1WayvM4DZLwE jelle@Jelles-Macbook-Pro.local"
+        "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAID+/7ktPyg4lYL0b6j3KQqfVE6rGLs5hNK3Q175th8cq jelle@foon"
+      ];
+    };
+    users.users.wouter = {
+      isNormalUser = true;
+      description = "Wouter Doeland";
+      extraGroups = [ "wheel" ];
+      openssh.authorizedKeys.keys = [
+        "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEblHIN5uaooHczkiqbXa6V7H7bfhgGTVLKA0sUggBkP wouter@wouterdoeland.nl"
+      ];
+    };
+
+    # This user is used by the GitHub Action to place the new configuration
+    users.users.deploy = {
+      isNormalUser = true;
+      description = "Deploy user";
+      extraGroups = [ "wheel" ];
+      openssh.authorizedKeys.keys = [
+        "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICxVTPZp7bOJhmU3hsK6yCSDALTC68/sQ5NRce531XGK deploykey"
+      ];
+    };
+
+    # Enable the SSH server, this also opens port 22 automatically
+    services.openssh.enable = true;
+
+    # Make concrexit user
+    users.users.${cfg.user} = { };
+
+    systemd.services = {
+      # Create the directory that concrexit uses to place files and the secrets
+      # This directory is created by this seperate service because the main service runs as the
+      # concrexit user
+      concrexit-dir = {
+        wantedBy = [ "multi-user.target" ];
+        script = ''
+          mkdir --parents ${cfg.dir}/media
+          chown ${cfg.user} ${cfg.dir}/media
+        '';
+      };
+
+      # The main systemd service which runs concrexit via uwsgi
+      concrexit = {
+        after = [ "networking.target" "postgresql.service" "concrexit-dir.service" ];
+        wantedBy = [ "multi-user.target" ];
+
+        serviceConfig = {
+          User = cfg.user;
+          KillSignal = "SIGQUIT";
+        };
+
+        script = ''
+          if [ -f ${cfg.dir}/secrets.env ]; then
+            source ${cfg.dir}/secrets.env
+          else
+            export DJANGO_SECRET=$(hostid)
+            echo "You should set the secrets in the env file!" >&2
+          fi
+
+          export STATIC_ROOT="${pkgs.concrexit}/static"
+          export SITE_DOMAIN="${if cfg.ssl then cfg.domain else "*"}"
+
+          # Load extra variables from the concrexit.env-vars option
+          ${concatStringsSep "\n" (mapAttrsToList (name: value: "export ${name}=\"${value}\"") cfg.env-vars)}
+
+          ${pkgs.concrexit}/bin/concrexit-uwsgi --socket :${toString cfg.app-port}
+        '';
+      };
+    };
+
+    # These configuration options are defined and implemented by the timed-commands.nix file
+    concrexit-timers = {
+      send_scheduled_messages = {
+        every = 60 * 5;
+        description = "Send scheduled push notifications";
+      };
+      sendplannednewsletters.every = 60 * 60;
```

Comment:
I think we should add some more descriptions

Reply:
I should add an issue to ask someone to explain what these management commands do

Reply:
Yes good idea